packages/grid/veilid/server/veilid_streamer.py

```diff
@@ -0,0 +1,477 @@
+            stream_single_request_header = self.stream_single_struct.pack(
+                RequestType.STREAM_SINGLE, call_id
+            )
+            stream_single_request = stream_single_request_header + message
+            await self._send_request(router, dht_key, stream_single_request)
+        else:
+            message_hash = hashlib.sha256(message).digest()
+            total_chunks_count = self._calculate_chunks_count(message_size)
+
+            # Send STREAM_START request
+            stream_start_request = self.stream_start_struct.pack(
+                RequestType.STREAM_START,
+                call_id,
+                message_hash,
+                total_chunks_count,
+            )
+            await self._send_request(router, dht_key, stream_start_request)
+
+            # Send chunks
+            tasks = []
+            for chunk_number in range(total_chunks_count):
+                chunk = self._get_chunk(call_id, chunk_number, message)
+                tasks.append(self._send_request(router, dht_key, chunk))
+            await asyncio.gather(*tasks)
+
+            # Send STREAM_END request
+            stream_end_message = self.stream_end_struct.pack(
+                RequestType.STREAM_END, call_id
+            )
+            await self._send_request(router, dht_key, stream_end_message)
+
+        if is_request_stream:
+            # This is a new request stream, so we need to wait for
+            # the reply from the receiver
+            logger.debug("Waiting for reply...")
+            response = await buffer_for_holding_reply.message
+            logger.debug("Reply received")
+            # All operations finished, clean up the buffer and return the response
+            del self.buffers[call_id]
+            return response
+        return ResponseType.OK
+
+    async def receive_stream(
+        self,
+        connection: veilid.VeilidAPI,
+        router: veilid.RoutingContext,
+        update: veilid.VeilidUpdate,
+        callback: AsyncReceiveStreamCallback,
+    ) -> None:
+        """Receives a streamed message."""
+        message = update.detail.message
+        prefix = message[:8]
+
+        if prefix == RequestType.STREAM_SINGLE:
+            await self._handle_receive_stream_single(
+                connection, router, update, callback
+            )
+        elif prefix == RequestType.STREAM_START:
+            await self._handle_receive_stream_start(connection, update)
+        elif prefix == RequestType.STREAM_CHUNK:
+            await self._handle_receive_stream_chunk(connection, update)
+        elif prefix == RequestType.STREAM_END:
+            await self._handle_receive_stream_end(connection, router, update, callback)
+        else:
+            logger.error(f"[Bad Message] Message with unknown prefix: {prefix}")
+
+    @retry(veilid.VeilidAPIError, tries=4, delay=1, backoff=2)
+    async def _send_request(
+        self, router: veilid.RoutingContext, dht_key: str, request_data: bytes
```

Comment:
MAYBE: Could we re-name in all the places from dht_key to vld_key with the new notation

Reply:
for the streamer part